src/input/element-input.js

```diff
@@ -842,11 +877,13 @@ class ElementInput {
                     }
                 }
                 ray = ray3d;
-            }
 
-            if (ray && this._checkElement(ray, element, screen)) {
-                result = element;
-                break;
+                // continue looking through all elements, only storing the closest one
+                const currentDistance = this._checkElement(ray, element, screen);
```

Comment:
this line needs to be wrapped in an if (ray)

Reply:
Good catch - and actually since the logic for calling `_checkElement` was moved to inside the ifs, the variable `ray` is no longer needed, and the code can directly use either `ray3d` or `rayScreen`. Fixed on next commit.